great_expectations/data_context/data_context/context_factory.py

```diff
@@ -84,37 +85,28 @@ def get_project(  # noqa: PLR0913
     def set_project(self, project: AbstractDataContext) -> None:
         self._project = project
 
+    def get_datasource_store(self) -> DatasourceStore:
+        return self._get_project_or_raise()._datasource_store
```

Comment:
i think these might be unnecessary, and worth getting rid of soon: https://greatexpectationslabs.slack.com/archives/C05V0M18TEJ/p1702307912270099?thread_ts=1702063530.058809&cid=C05V0M18TEJ